fideslib/core/config.py

```diff
@@ -0,0 +1,255 @@
+
+    class Config:
+        env_prefix = "FIDES__SECURITY__"
+
+
+class FidesConfig(FidesSettings):
+    """Configuration variables for the FastAPI project."""
+
+    database: DatabaseSettings
+    security: SecuritySettings
+
+    is_test_mode: bool = os.getenv("TESTING") == "True"
+    hot_reloading: bool = os.getenv("FIDES__HOT_RELOAD") == "True"
+    dev_mode: bool = os.getenv("FIDES__DEV_MODE") == "True"
+
+    class Config:
+        case_sensitive = True
+
+
+def load_file(file_names: Union[List[Path], List[str]]) -> str:
+    """Load a file from the first matching location.
+
+    In order, will check:
+    - A path set at ENV variable FIDES__CONFIG_PATH
+    - The current directory
+    - The parent directory
+    - The parent_directory/.fides
+    - users home (~) directory
+    raises FileNotFound if none is found
+    """
+
+    possible_directories = [
+        os.getenv("FIDES__CONFIG_PATH"),
+        os.curdir,
+        os.pardir,
+        os.path.join(os.pardir, ".fides"),
+        os.path.expanduser("~"),
+    ]
+
+    directories = [d for d in possible_directories if d]
+
+    for dir_str in directories:
+        for file_name in file_names:
+            possible_location = os.path.join(dir_str, file_name)
+            if possible_location and os.path.isfile(possible_location):
+                logger.info(
+                    "Loading file %s from %s", NotPii(file_name), NotPii(dir_str)
+                )
+                return possible_location
+
+    raise FileNotFoundError
+
+
+def load_toml(file_names: Union[List[Path], List[str]]) -> MutableMapping[str, Any]:
+    """Load toml file from possible locations specified in load_file.
+
+    Will raise FileNotFoundError or ValidationError on missing or
+    bad file
+    """
+    file_name = load_file(file_names)
+    with open(file_name, "rb") as f:
+        return tomli.load(f)
+
+
+def get_config(
+    class_name: Type[FidesConfig] = FidesConfig,
+    *,
+    file_names: Union[List[Path], List[str]] = [
+        "fidesops.toml",
+        "fidesctl.toml",
+        "fides.toml",
+    ],
+) -> FidesConfig:
+    """
+    Attempt to read config file named fidesops.toml, fidesctl.toml, or fides.toml from:
+    - env var FIDES__CONFIG_PATH
+    - local directory
+    - parent directory
+    - home directory
+    This will fail on the first encountered bad conf file.
+    """
+
+    try:
+        return class_name.parse_obj(load_toml(file_names))
+    except (FileNotFoundError, ValidationError) as e:
+        if isinstance(file_names, list):
+            if len(file_names) == 1:
+                logger.warning("%s could not be loaded: %s", file_names[0], NotPii(e))
+            else:
+                logger.warning(
+                    "%s could not be loaded: %s",
+                    " or ".join([str(x) for x in file_names]),
+                    NotPii(e),
+                )
+        else:
+            logger.warning("%s could not be loaded: %s", file_names, NotPii(e))
```

Comment:
`file_names` will always be an instance of `list` now.
```suggestion
        if len(file_names) == 1:
            logger.warning("%s could not be loaded: %s", file_names[0], NotPii(e))
        else:
            logger.warning(
                "%s could not be loaded: %s",
                " or ".join([str(x) for x in file_names]),
                NotPii(e),
            )
```

Reply:
Yes, good catch. I missed this one when updating.